crates/swift-bridge-ir/src/bridged_type.rs

```diff
@@ -481,6 +481,7 @@ impl BridgeableType for BridgedType {
 
     fn is_passed_via_pointer(&self) -> bool {
         match self {
+            BridgedType::StdLib(StdLibType::Vec(_)) => true,
```

Comment:
If we use `Result<Vec<RustOpaqueType>, RustOpaqueType>`, we need to test that `__private__ResultPtrAndPtr` is generated, not a custom result type.

Reply:
As mentioned in my review I think this wouldn't need a codegen test but maybe instead we'd have some tests in the `BuiltInResult` file where we check which FFI representation is used by different `Result<T, E>` signatures.

So, we'd essentially be testing this function https://github.com/chinedufn/swift-bridge/blob/637c7b353539c1122dacc3a64e7d5dc052a4fec9/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs#L395

Doesn't need to happen in this PR though.

Thanks a lot for reviewing this PR, very helpful!!! Thank you!